dev/tools/check-eof-newline.sh

```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-if [ -z "$(tail -c 1 "$1")" ]
+if ! dev/tools/should-check-whitespace.sh "$1" || [ -z "$(tail -c 1 "$1")" ]
```

Comment:
Didn't you remove this file in another PR of yours?

Reply:
>Didn't you remove this file in another PR of yours?

#6568 which isn't merged yet.
This PR and that one conflict so I'll just fix whichever gets merged second.